### DISCUSSION TITLE
Deepmd installation error

### DISCUSSION THREAD

**Discussion by Lewis-YL** [deepmodeling/deepmd-kit#1778] [Q&A] [answered]
Hello,

I am trying to install deepmd 2.1.1 on the HPC cluster.

However, there is some error:

I used the command:

conda create -n deepmd deepmd-kit=2.1.1=*gpu libdeepmd=2.1.1=*gpu lammps-dp cudatoolkit=11.3 horovod -c https://conda.deepmodeling.org

It produced error:

Collecting package metadata (repodata.json): done
Solving environment: - 
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed                                                                                                                                    / 

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package openssl conflicts for:
libdeepmd==2.1.1[build=*gpu] -> libtensorflow_cc=2.9[build=cuda*] -> openssl[version='>=1.1.1l,<1.1.2a|>=1.1.1o,<1.1.2a']
deepmd-kit==2.1.1[build=*gpu] -> python[version='>=3.8,<3.9.0a0'] -> openssl[version='>=1.0.2o,<1.0.3a|>=1.0.2p,<1.0.3a|>=1.1.1a,<1.1.2a|>=1.1.1b,<1.1.2a|>=1.1.1c,<1.1.2a|>=1.1.1d,<1.1.2a|>=1.1.1e,<1.1.2a|>=1.1.1g,<1.1.2a|>=1.1.1j,<1.1.2a|>=1.1.1k,<1.1.2a|>=1.1.1l,<1.1.2a|>=1.1.1n,<1.1.2a|>=1.1.1o,<1.1.2a|>=1.1.1i,<1.1.2a|>=1.1.1h,<1.1.2a|>=1.1.1m,<1.1.2a']
horovod -> python[version='>=3.9,<3.10.0a0'] -> openssl[version='>=1.0.2o,<1.0.3a|>=1.0.2p,<1.0.3a|>=1.1.1a,<1.1.2a|>=1.1.1b,<1.1.2a|>=1.1.1c,<1.1.2a|>=1.1.1d,<1.1.2a|>=1.1.1e,<1.1.2a|>=1.1.1g,<1.1.2a|>=1.1.1h,<1.1.2a|>=1.1.1i,<1.1.2a|>=1.1.1j,<1.1.2a|>=1.1.1k,<1.1.2a|>=1.1.1l,<1.1.2a|>=1.1.1n,<1.1.2a|>=1.1.1o,<1.1.2a|>=1.1.1m,<1.1.2a']

Package cudatoolkit conflicts for:
horovod -> cudatoolkit[version='*|10.1|10.1.*|10.2|10.2.*|>=11.6,<12|>=11.3,<12|>=11.3.1,<11.4.0a0*|>=10.1.243,<10.2.0a0*']
horovod -> tensorflow-base=2.5.0[build=gpu*] -> cudatoolkit[version='10.0|10.0.*|10.1.*|11.3.*|9.2|9.2.*']
lammps-dp -> libdeepmd[version='>=2.0.2'] -> cudatoolkit[version='10.1|10.1.*|10.2|10.2.*|>=11.6,<12|>=11.3,<12|>=11.3.1,<11.4.0a0|>=11.1.74,<11.2.0a0|>=11.0.221,<11.1.0a0']
lammps-dp -> cudatoolkit[version='>=10.0.130,<10.1.0a0|>=10.1.243,<10.2.0a0|>=9.2,<9.3.0a0']
deepmd-kit==2.1.1[build=*gpu] -> cudatoolkit[version='10.1|10.1.*|10.2|10.2.*|>=11.6,<12|>=11.3,<12']
cudatoolkit=11.3
libdeepmd==2.1.1[build=*gpu] -> cudatoolkit[version='10.1|10.1.*|10.2|10.2.*|>=11.6,<12|>=11.3,<12']

Package mpi conflicts for:
horovod -> mpich[version='>=3.3.2,<3.4.0a0'] -> mpi==1.0=mpich
libdeepmd==2.1.1[build=*gpu] -> mpich[version='>=3.3.2,<3.4.0a0'] -> mpi==1.0=mpich
lammps-dp -> mpich[version='>=3.3.2,<3.4.0a0'] -> mpi==1.0=mpich

Package libdeepmd conflicts for:
libdeepmd==2.1.1[build=*gpu]
lammps-dp -> liblammps-dp==1.3.3=1_cuda9.2_gpu -> libdeepmd[version='1.2.2|1.2.2|1.2.2|1.2.2|1.2.2|1.2.2|1.2.2|1.2.2|1.2.2|1.2.2|1.2.2|1.2.2|1.2.2|1.2.2|1.2.2|1.2.2|1.2.3|1.2.3|1.2.3|1.2.3|1.2.3|1.2.3|1.2.3|1.2.3|1.2.4|1.2.4|1.2.4|1.2.4|1.2.4|1.2.4|1.2.4|1.2.4|1.3.0|1.3.0|1.3.0|1.3.0|1.3.0|1.3.0|1.3.0|1.3.0|1.3.1|1.3.1|1.3.1|1.3.1|1.3.1|1.3.1|1.3.1|1.3.1|1.3.2|1.3.2|1.3.2|1.3.2|1.3.2|1.3.2|1.3.2|1.3.2|1.3.2|1.3.2|1.3.2|1.3.2|1.3.3',build='0_cuda9.2_gpu_low|0_cuda9.2_gpu|0_cuda10.0_gpu|1_cudaNone_cpu_low|1_cuda9.2_gpu_low|1_cuda9.2_gpu|0_cudaNone_cpu|0_cuda9.2_gpu_low|0_cuda10.1_gpu|0_cuda10.1_gpu_low|0_cuda9.2_gpu|0_cuda10.0_gpu|0_cuda10.1_gpu_low|0_cuda10.0_gpu|0_cudaNone_cpu|0_cuda9.2_gpu_low|0_cuda9.2_gpu|0_cuda10.1_gpu|0_cuda9.2_gpu|0_cuda9.2_gpu_low|0_cuda10.0_gpu_low|0_cuda9.2_gpu_low|0_cuda9.2_gpu|0_cuda10.0_gpu|0_cuda10.0_gpu_low|0_cuda10.1_gpu|0_cuda9.2_gpu|0_cuda9.2_gpu_low|0_cuda11.1_gpu_low|0_cuda11.0_gpu|0_cuda10.1_gpu|0_cudaNone_cpu|0_cuda9.2_gpu_low|0_cuda11.0_gpu|0_cuda11.1_gpu_low|0_cuda9.2_gpu|0_cuda11.1_gpu|0_cudaNone_cpu_low|0_cuda11.0_gpu_low|0_cuda10.1_gpu_low|0_cuda10.0_gpu|0_cuda10.0_gpu_low|0_cuda11.0_gpu_low|0_cuda11.1_gpu|0_cuda10.0_gpu_low|0_cuda10.0_gpu|0_cuda10.1_gpu_low|0_cudaNone_cpu|0_cudaNone_cpu_low|0_cuda10.1_gpu_low|0_cuda10.1_gpu|0_cudaNone_cpu_low|0_cudaNone_cpu|0_cuda10.0_gpu|0_cuda10.1_gpu_low|0_cudaNone_cpu|0_cudaNone_cpu_low|0_cudaNone_cpu_low|0_cuda10.0_gpu_low|0_cuda10.1_gpu|0_cuda10.0_gpu_low|0_cudaNone_cpu_low|1_cuda10.0_gpu|1_cuda10.0_gpu_low|1_cuda10.1_gpu|1_cuda10.1_gpu_low|1_cudaNone_cpu|0_cuda10.1_gpu_low|0_cuda10.1_gpu|0_cuda10.0_gpu_low|0_cudaNone_cpu|0_cudaNone_cpu_low']
lammps-dp -> libdeepmd[version='1.0.1|1.0.1|1.0.1|1.0.1|1.1.0|1.1.0|1.1.0|1.1.0|1.1.1|1.1.1|1.1.1|1.1.1|1.1.2|1.1.2|1.1.2|1.1.2|1.1.3|1.1.3|1.1.3|1.1.3|1.1.3|1.1.3|1.1.3|1.1.3|1.1.3|1.1.3|1.1.3|1.1.3|1.1.4|1.1.4|1.1.4|1.1.4|1.1.4|1.1.4|1.1.4|1.1.4|1.1.5|1.1.5|1.1.5|1.1.5|1.1.5|1.1.5|1.1.5|1.1.5|1.2.0|1.2.0|1.2.0|1.2.0|1.2.0|1.2.0|1.2.0|1.2.0|>=2.0.0|>=2.0.1|>=2.0.2',build='*cpu|*gpu_low|*cpu_low|*cpu_low|*cpu|*gpu_low|*cpu_low|*cpu|*gpu_low|*cpu_low|*cpu|*gpu_low|*cpu_low|*gpu_low|1_cudaNone_cpu_low|1_cuda9.2_gpu_low|1_cuda9.2_gpu|1_cuda10.0_gpu|1_cuda10.1_gpu|0_cuda9.2_gpu_low|0_cuda10.0_gpu_low|0_cuda9.2_gpu|0_cuda9.2_gpu|0_cuda9.2_gpu_low|0_cuda10.1_gpu|0_cuda10.1_gpu_low|0_cuda9.2_gpu|0_cuda9.2_gpu_low|0_cuda10.1_gpu|0_cuda10.0_gpu_low|0_cuda10.1_gpu_low|0_cuda10.0_gpu|0_cudaNone_cpu|0_cudaNone_cpu_low|0_cuda10.0_gpu_low|0_cuda10.0_gpu|0_cudaNone_cpu_low|0_cudaNone_cpu|0_cuda10.0_gpu|0_cuda10.1_gpu_low|0_cuda10.1_gpu|0_cudaNone_cpu_low|0_cudaNone_cpu|1_cuda10.1_gpu_low|1_cuda10.0_gpu_low|1_cudaNone_cpu|*gpu|*cpu|*gpu|*gpu|*gpu|*gpu']

Package keras conflicts for:
deepmd-kit==2.1.1[build=*gpu] -> tensorflow=2.9[build=cuda*] -> keras[version='>=2.7,<2.8|>=2.9,<2.10']
horovod -> tensorflow=2.9[build=cuda*] -> keras[version='>=2.7,<2.8|>=2.9,<2.10']

Package cudnn conflicts for:
horovod -> tensorflow=2.9[build=cuda*] -> cudnn[version='7.*|>=7.6,<8.0a0|>=8.4.0.27,<9.0a0|>=8.2,<9.0a0|8.*']
lammps-dp -> libdeepmd[version='>=2.0.2'] -> cudnn[version='>=7.6,<8.0a0|>=8.2,<9.0a0|>=8.2.1,<9.0a0|>=7.6.5,<8.0a0|>=7.6.4,<8.0a0|>=7.6.0,<8.0a0']
deepmd-kit==2.1.1[build=*gpu] -> tensorflow=2.9[build=cuda*] -> cudnn[version='>=7.6,<8.0a0|>=8.4.0.27,<9.0a0|>=8.2,<9.0a0']

Package mpi4py conflicts for:
lammps-dp -> pylammps-dp==2.0.2=pyh6e9818e_6 -> mpi4py
horovod -> mpi4py

Package mpich conflicts for:
lammps-dp -> mpich[version='>=3.2,<3.3.0a0|>=3.3.2,<3.4.0a0|>=3.3.2,<4.0a0|>=3.2.1,<4.0a0']
lammps-dp -> pylammps-dp==2.0.2=pyh6e9818e_6 -> mpich

Package deepmd-kit conflicts for:
deepmd-kit==2.1.1[build=*gpu]
lammps-dp -> deepmd-kit==0.12.8[build='*cpu|*gpu']

Package nccl conflicts for:
libdeepmd==2.1.1[build=*gpu] -> nccl[version='>=2.11.4.1,<3.0a0|>=2.12.12.1,<3.0a0']
horovod -> nccl[version='>=2.11.4.1,<3.0a0|>=2.12.12.1,<3.0a0|>=2.8.3.1,<3.0a0']
lammps-dp -> libdeepmd[version='>=2.0.2'] -> nccl[version='>=2.11.4.1,<3.0a0|>=2.12.12.1,<3.0a0']
deepmd-kit==2.1.1[build=*gpu] -> tensorflow=2.9[build=cuda*] -> nccl[version='>=2.11.4.1,<3.0a0|>=2.12.12.1,<3.0a0']

Package zlib conflicts for:
lammps-dp -> zlib[version='>=1.2.11,<1.3.0a0']
lammps-dp -> plumed[version='>=2'] -> zlib[version='>=1.2.12,<1.3.0a0']The following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.17=0
  - feature:|@/linux-64::__glibc==2.17=0
  - deepmd-kit==2.1.1[build=*gpu] -> __glibc[version='>=2.17']
  - deepmd-kit==2.1.1[build=*gpu] -> cudatoolkit[version='>=11.6,<12'] -> __glibc[version='>=2.17,<3.0.a0']
  - horovod -> __cuda
  - horovod -> __glibc[version='>=2.17']
  - horovod -> cudatoolkit[version='>=11.6,<12'] -> __glibc[version='>=2.17,<3.0.a0']
  - lammps-dp -> libdeepmd[version='>=2.0.2'] -> __glibc[version='>=2.17']
  - libdeepmd==2.1.1[build=*gpu] -> __glibc[version='>=2.17']
  - libdeepmd==2.1.1[build=*gpu] -> cudatoolkit[version='>=11.6,<12'] -> __glibc[version='>=2.17,<3.0.a0']

Your installed version is: 2.17

Do you know how to solve the problem?
Thank you!

**Comment by njzjz** [comment, marked as answer]
See #1362
